### DISCUSSION TITLE
Reduce API calls - insert statement through the client or through a postgres function

### DISCUSSION THREAD

**Discussion by sumchans** [supabase/supabase#3027] [Questions] [answered]
I am a beginner programmer. Just want to clarify this before I continue on whatever path I choose. The whole objective is to reduce the number of API calls.

I want to do insert statements to two different tables in postgres. The second insert statement should only happen if a record exists in the first table.

So if I do it through the client, it would involve two API calls, each time I do an insert to the second table.
1. I have to check if the record is there in the first table
2. If the record exists insert data into second table.

So if I am doing the above insert statements using a function 
1. I can combine both the insert statements onto a function and make single call to the function from my client and this would involve only one API call.

Anybody with expertise please confirm if I am thinking the right way.

**Comment by w3b6x9** [comment, marked as answer]
@sumchans i think your intuition is correct and using a function is preferable given your case and what you're optimizing for.

**Reply by sumchans** [reply]
@w3b6x9 Thanks Wen bo.
